src/server/routes/index.ts

```diff
@@ -47,8 +48,10 @@ uncachedRoutes.use(verifyEmail);
 uncachedRoutes.use(magicLink);
 
 // welcome routes
-uncachedRoutes.use(noCache, welcome);
+uncachedRoutes.use(welcome);
 
 router.use(uncachedRoutes);
 
+router.use(notFound);
```

Comment:
if res.locals gets passed to 404 then this should be uncached to avoid leaking sensitive data 🤔

Reply:
Good spot!